src/eth/storage/overlay_storage.rs

```diff
@@ -0,0 +1,65 @@
+use async_trait::async_trait;
+
+use crate::eth::primitives::Account;
+use crate::eth::primitives::Address;
+use crate::eth::primitives::BlockNumber;
+use crate::eth::primitives::Execution;
+use crate::eth::primitives::ExecutionConflicts;
+use crate::eth::primitives::Slot;
+use crate::eth::primitives::SlotIndex;
+use crate::eth::primitives::StoragePointInTime;
+
+/// Temporary storage (in-between blocks) operations
+// TODO: add Metrified method
+#[async_trait]
+pub trait OverlayStorage: Send + Sync {
```

Comment:
I think `OverlayStorage` is not descriptive enough (or at all) to understand the purpose of this struct.

The name in the comments (`InBetweenBlocksStorage` or `TemporaryStorage`) or something else can give a better view of what is the purpose.

Reply:
fair enough, I'll rename it to TemporaryStorage